lib/knapsack_pro/extensions/rspec_extension.rb

```diff
@@ -0,0 +1,164 @@
+        end
+
+        def knapsack__exit_early
+          _exit_status = configuration.reporter.exit_early(exit_code)
+        end
+
+        # Based on:
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/configuration.rb#L546
+        # Set with --error-exit-code CODE - Override the exit code used when there are errors loading or running specs outside of examples.
+        def knapsack__error_exit_code
+          _default = nil || configuration.error_exit_code
+        end
+
+        def knapsack__deprecated_run_all_when_everything_filtered_enabled?
+          raise "Cannot call the #{__method__} method because setup was not done." unless knapsack__setup_done?
+
+          !!(configuration.respond_to?(:run_all_when_everything_filtered) && configuration.run_all_when_everything_filtered)
+        end
+
+        def knapsack__seed
+          configuration.seed.to_s
+        end
+
+        def knapsack__seed_used?
+          configuration.seed_used?
+        end
+
+        # @param test_file_paths Array[String]
+        #   Example: ['a_spec.rb', 'b_spec.rb[1:1]']
+        def knapsack__load_spec_files_batch(test_file_paths)
+          world.reset
+
+          # Reset filters, but do not reset `configuration.static_config_filter_manager` to preserve the --tag option
+          filter_manager = RSpec::Core::FilterManager.new
+          options.configure_filter_manager(filter_manager)
+          configuration.filter_manager = filter_manager
+
+          configuration.knapsack__load_spec_files(test_file_paths)
+        end
+
+        # Based on:
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/runner.rb#L113
+        #
+        # Option: `configuration.fail_if_no_examples`
+        #   Ignore the configuration.fail_if_no_examples option because it should be off in Queue Mode.
+        #   * Ignore the fail_if_no_examples option because in Queue Mode a late CI node can start after other CI nodes already executed tests. It is expected not to run examples in such a scenario.
+        #   * RSpec should not fail when examples are not executed for a batch of tests fetched from the Queue API. The batch could have tests that have no examples (for example, someone commented out the content of the spec file). We should fetch another batch of tests from the Queue API and keep running tests.
+        #
+        # @return [Fixnum] exit status code.
+        def knapsack__run_specs(queue_runner)
+          # Based on:
+          # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/world.rb#L53
+          ordering_strategy = configuration.ordering_registry.fetch(:global)
+          node_examples_passed = true
+
+          configuration.with_suite_hooks do
+            exit_status = configuration.reporter.report(_expected_example_count = 0) do |reporter|
```

Comment:
Indeed. The code was proposed in this order based on https://github.com/KnapsackPro/knapsack_pro-ruby/pull/226

Since we have integration tests now, I replaced the order and tests pass. We can do it in the same order as in the original code: https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/runner.rb#L115

Fix: https://github.com/KnapsackPro/knapsack_pro-ruby/pull/237/commits/7c782a8b48cde3e8a017669970d282f9feec97ef